### ISSUE TITLE
Whitespace-only input crashes xonsh

### ISSUE THREAD

**Issue by mbollmann** [scopatz/xonsh#32]
Entering only spaces and pressing enter crashes xonsh:

```
bollmann@bollmann-VirtualBox ~ >>>  
Traceback (most recent call last):
  File "/home/bollmann/repositories/xonsh/xonsh/execer.py", line 118, in _parse_ctx_free
    mode=mode, debug_level=self.debug_level)
  File "/home/bollmann/repositories/xonsh/xonsh/parser.py", line 271, in parse
    debug=debug_level)
  File "/usr/local/lib/python3.4/dist-packages/ply-3.4-py3.4.egg/ply/yacc.py", line 265, in parse
  File "/usr/local/lib/python3.4/dist-packages/ply-3.4-py3.4.egg/ply/yacc.py", line 1047, in parseopt_notrack
  File "/home/bollmann/repositories/xonsh/xonsh/parser.py", line 2120, in p_error
    column=self.lexer.token_col(p)))
  File "/home/bollmann/repositories/xonsh/xonsh/parser.py", line 341, in _parse_error
    raise err
SyntaxError: <xonsh-code>:1:1: ('code:  ',)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/xonsh", line 12, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/home/bollmann/repositories/xonsh/scripts/xonsh", line 3, in <module>
    main()
  File "/home/bollmann/repositories/xonsh/xonsh/main.py", line 15, in main
    shell.cmdloop()
  File "/home/bollmann/repositories/xonsh/xonsh/shell.py", line 144, in cmdloop
    super(Shell, self).cmdloop(intro=intro)
  File "/usr/lib/python3.4/cmd.py", line 138, in cmdloop
    stop = self.onecmd(line)
  File "/usr/lib/python3.4/cmd.py", line 211, in onecmd
    return self.default(line)
  File "/home/bollmann/repositories/xonsh/xonsh/shell.py", line 97, in default
    code = self.push(line)
  File "/home/bollmann/repositories/xonsh/xonsh/shell.py", line 118, in push
    locs=self.ctx)
  File "/home/bollmann/repositories/xonsh/xonsh/execer.py", line 84, in compile
    tree = self.parse(input, ctx, mode=mode)
  File "/home/bollmann/repositories/xonsh/xonsh/execer.py", line 62, in parse
    tree = self._parse_ctx_free(input, mode=mode)
  File "/home/bollmann/repositories/xonsh/xonsh/execer.py", line 126, in _parse_ctx_free
    lines[idx] = subproc_line(lines[idx])
  File "/home/bollmann/repositories/xonsh/xonsh/tools.py", line 31, in subproc_line
    tok = line.split(None, 1)[0]
IndexError: list index out of range
Exception ignored in: <bound method Shell.__del__ of <xonsh.shell.Shell object at 0x7ff7874cb8d0>>
Traceback (most recent call last):
  File "/home/bollmann/repositories/xonsh/xonsh/shell.py", line 83, in __del__
  File "/home/bollmann/repositories/xonsh/xonsh/shell.py", line 49, in teardown_readline
  File "<frozen importlib._bootstrap>", line 2214, in _find_and_load
  File "<frozen importlib._bootstrap>", line 2199, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 2132, in _find_spec
TypeError: 'NoneType' object is not iterable
```

The same happens in multi-line input, e.g. if you separate parts of an indented block (say, in .xonshrc) with an empty line, and that line happens to contain whitespace.

**Comment by scopatz** [comment]
Now whitespace only input never terminates, but neither does to it enter multiline mode. definitely a bug. Thanks for reporting.

**Comment by scopatz** [comment]
Somewhat hilariously, this causes an infinite loop.

**Comment by scopatz** [comment]
This should be fixed with 2a4d126 @mbollmann. Can you confirm?

**Comment by mbollmann** [comment]
Yes, works now.
